src/Features/LanguageServer/ProtocolUnitTests/Ordering/LongRunningNonMutatingRequestHandler.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Xunit.Sdk;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.RequestOrdering
+{
+    [Shared, ExportLspRequestHandlerProvider, PartNotDiscoverable]
+    [ProvidesMethod(LongRunningNonMutatingRequestHandler.MethodName)]
+    internal class LongRunningNonMutatingRequestHandlerProvider : AbstractRequestHandlerProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public LongRunningNonMutatingRequestHandlerProvider()
+        {
+        }
+
+        public override ImmutableArray<IRequestHandler> CreateRequestHandlers() => ImmutableArray.Create<IRequestHandler>(new LongRunningNonMutatingRequestHandler());
+    }
+
+    internal class LongRunningNonMutatingRequestHandler : IRequestHandler<TestRequest, TestResponse>
+    {
+        public const string MethodName = nameof(LongRunningNonMutatingRequestHandler);
+
+        public string Method => MethodName;
+
+        public bool MutatesSolutionState => false;
+
+        public bool RequiresLSPSolution => true;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(TestRequest request) => null;
+
+        public Task<TestResponse> HandleRequestAsync(TestRequest request, RequestContext context, CancellationToken cancellationToken)
+        {
+            do
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return Task.FromResult(new TestResponse());
+                }
+
+                Thread.Sleep(100);
+            } while (true);
```

Comment:
My first response was "No way" but I see this is test code where you're just trying to force the wait.

Reply:
Yup, can't do `await Task.Delay(...)` because that'd yield 😄. Key portion of this test code is that it's doing a synchronous operation and consuming a thread for the entirety of the test